Made refresh button rotate while repositories are loading to be viewed

CabbitKheema opened the pull request:
This PR is the solution for #4 

Used `RefreshCw` component instead of `RefreshCcw` from `lucide-react` for two reasons 
- Reason 1: This follows common UI conventions seen in browsers, apps, and operating systems 
- Reason 2: Tailwind Css's `animate-spin` naturally rotates clockwise and using`RefreshCw` component supports this without additional changes.

**Result**
Now the refresh button rotates whenever repositories are loaded to be viewed

https://github.com/user-attachments/assets/b4d47319-c657-46b4-8dda-4b54efec85e8

Thank you for your time! 😄

CabbitKheema committed:
Made refresh button rotate while repositories are loaded

- Used 'RefreshCw' component instead of 'RefreshCcw' from 'lucide-react' for two reasons
Reason 1: This follows common UI conventions seen in browsers, apps, and operating systems
Reason 2: Tailwind Css's 'animate-spin' naturally rotates clockwise and using 'RefreshCw' component supports this without additional changes.

nilbuild commented on src/components/Navigation.tsx:
```diff
@@ -31,7 +32,7 @@ export function Navigation(props: NavigationProps) {
             className='p-2 text-white/70 hover:text-white transition-colors'
           >
-            <RefreshCcw className='w-5 h-5' />
+            <RefreshCw className={`w-5 h-5 ${dataLoading ? 'animate-spin' : ''}`} />
```
Let's also disable the button while loading is in progress

CabbitKheema replied:
Sure thing.

CabbitKheema replied:
@kamranahmedse 
Is there anything else you'd like me to do? 🤔

CabbitKheema committed:
Disabled the refresh button while loading is in progress

CabbitKheema commented:
@kamranahmedse 
Is there anything else you'd like me to do? 🤔

CabbitKheema committed:
Changed flag variable name from "dataLoading" to "isDataLoading"

- This change supports the convention for naming flag variables of boolean type.

CabbitKheema commented:
The flag variable renaming felt necessary

@kamranahmedse
Is there anything else you'd like me to do? 🤔
